src/monodroid/jni/monovm-properties.cc

```diff
@@ -5,9 +5,11 @@ using namespace xamarin::android::internal;
 MonoVMProperties::property_array MonoVMProperties::_property_keys {
 	RUNTIME_IDENTIFIER_KEY,
 	APP_CONTEXT_BASE_DIRECTORY_KEY,
+	LOCAL_DATE_TIME_OFFSET_KEY,
```

Comment:
I hope that one day we can replace this dynamically allocated property with a field in the `MonoCoreRuntimeProperties` struct... It would make the code even faster.

Reply:
How much are we saving if we could leverage `MonoCoreRuntimeProperties` struct over adding another property key/value pair?

There isn't any convenient way to put any of the `MonoCoreRuntimeProperties` into AppContext on the runtime side. This is because none of the properties in `MonoCoreRuntimeProperties` need to make it to the managed side. Another thing we would have to do is maybe differentiate between properties that need to make it to managed side or not to then specifically add into the AppContext.

Since populating AppContext is necessary for startup to finish, would the savings done by leveraging MonoCoreRuntimeProperties over adding a new key/value pair here be greater than the cost of needing to convert it to a string property in the runtime and any possible filtering before its added to the App Context?

Reply:
The biggest savings on XA side come from not having to convert an integer into a string and not duplicating that string with `strdup`. These obviously aren't big savings, but it's something that needlessly consumes memory and processor time (since that string will be parsed back into integer a short while later). I wonder if AppContext can be populated lazily? That is, could there be code to retrieve the value(s) from `MonoCoreRuntimeProperties` on demand (using an icall or p/invoke) instead of pre-populating it?

Also, looking at `AppContext` docs, its `GetData(string)` method returns an `object?` - do we absolutely have to convert the integer retrieved here to string, then use `Convert.ToInt32` after retrieving it from `GetData`?
This just looks like a wasteful way of doing it... After all, we're running in the same process, so we can access variables (sort of) directly without having to proxy them via strings.  There was a similar issue sometime ago with p/invoke overrides, where initially we had to convert a pointer to a hexadecimal string, pass it to the runtime which would then parse that string and cast it back into a pointer...

On desktop these string operations may not matter, on mobile it's an entirely different matter - every nanosecond saved counts.